eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -287,16 +294,19 @@ public function publish($contentId, $versionNo, MetadataUpdateStruct $struct)
         $content = $this->persistenceHandler->contentHandler()->publish($contentId, $versionNo, $struct);
 
         $this->cache->clear('content', $contentId);
+        $this->cache->clear('content', 'info', $contentId);
         $this->cache->clear('content', 'info', 'remoteId');
         $this->cache->clear('location', 'subtree');
 
         // warm up cache
-        $contentInfo = $content->versionInfo->contentInfo;
+        $versionInfo = $content->versionInfo;
+        $contentInfo = $versionInfo->contentInfo;
         $this->cache
             ->getItem('content', $contentInfo->id, $content->versionInfo->versionNo, self::ALL_TRANSLATIONS_KEY)
             ->set($content)
             ->save();
         $this->cache->getItem('content', 'info', $contentInfo->id)->set($contentInfo)->save();
+        $this->cache->getItem('content', 'info', $contentInfo->id, 'versioninfo', $contentInfo->currentVersionNo)->set($versionInfo)->save();
```

Comment:
Note: Kind of moving in direction of removing the cache warmers, there is no guarantee it will be used anytime soon after operation, so there is a risk they waste cache space.  Like in case of content cache above it is cached for all translations, while it will perhaps only be loaded in specific translations normally.

So ideally I would skip this if you agree.

Reply:
> So ideally I would skip this if you agree.

No problem.